app/controllers/hackdays_controller.rb

```diff
@@ -0,0 +1,29 @@
+class HackdaysController < ApplicationController
+
+  def index
+    @hackday = Hackday.new
+    @hackdays = Hackday.all
+  end
+
+  def show
+    @hackday = Hackday.find(params[:id])
+    @project = @hackday.projects.build
+  end
+
+  def create
+    @hackday = Hackday.new(:name => params[:hackday][:name])
+    if @hackday.save
+      redirect_to hackday_path(@hackday)
+    else
+      @hackdays = Hackday.all
+      render 'index'
```

Comment:
bad form to render index. especially since you had to basically repeat all the code in the index

you should just have another action with just the form